numpy/core/numeric.py

```diff
@@ -289,16 +288,14 @@ def full(shape, fill_value, dtype=None, order='C'):
     >>> np.full((2, 2), np.inf)
     array([[ inf,  inf],
            [ inf,  inf]])
-    >>> np.full((2, 2), 10, dtype=np.int)
+    >>> np.full((2, 2), 10)
     array([[10, 10],
            [10, 10]])
 
     """
+    if dtype is None:
+        dtype = array(fill_value).dtype
```

Comment:
Forget about this, I guess.... object dtype is funny, and you just have to do it manually. Even noticed we had talked about it before, heh. I guess you are right about `obj2sctype`, it might be trying to recreate the corresponding C-function, no idea....